From mtosatti@redhat.com  Thu Sep  3 14:24:08 2009
From: Amit Shah <amit.shah@redhat.com>
Date: Mon,  3 Aug 2009 14:57:55 -0300
Subject: KVM: x86: Ignore reads to EVNTSEL MSRs
To: stable@kernel.org
Cc: Amit Shah <amit.shah@redhat.com>, avi@redhat.com
Message-ID: <1249322277-5824-8-git-send-email-mtosatti@redhat.com>


From: Amit Shah <amit.shah@redhat.com>

(cherry picked from commit 7fe29e0faacb650d31b9e9f538203a157bec821d)

We ignore writes to the performance counters and performance event
selector registers already. Kaspersky antivirus reads the eventsel
MSR causing it to crash with the current behaviour.

Return 0 as data when the eventsel registers are read to stop the
crash.

Signed-off-by: Amit Shah <amit.shah@redhat.com>
Signed-off-by: Avi Kivity <avi@redhat.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 arch/x86/kvm/x86.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -898,6 +898,8 @@ int kvm_get_msr_common(struct kvm_vcpu *
 	case MSR_IA32_LASTINTFROMIP:
 	case MSR_IA32_LASTINTTOIP:
 	case MSR_VM_HSAVE_PA:
+	case MSR_P6_EVNTSEL0:
+	case MSR_P6_EVNTSEL1:
 		data = 0;
 		break;
 	case MSR_MTRRcap:
